src/util.c

```diff
@@ -540,12 +563,17 @@ All facets connected.  No further nearby check necessary.\n");
     printf("Calculating volume...\n");
   stl_calculate_volume(stl);
 
+  if (verbose_flag)
+    printf("Calculate surfacea area...\n");
+  stl_calculate_surface_area(stl);
+
   if(fixall_flag) {
     if(stl->stats.volume < 0.0) {
       if (verbose_flag)
         printf("Reversing all facets because volume is negative...\n");
       stl_reverse_all_facets(stl);
       stl->stats.volume = -stl->stats.volume;
+      stl->stats.surface_area = -stl->stats.surface_area;
```

Comment:
If the volume is negative, you set surface area to it's negative value. What is the reason? The surface are is not negative as well.